docs/user-manuals/node-resource-fit-plus-scoring.md

```diff
@@ -0,0 +1,237 @@
+# Enhanced NodeResourceFit Plugin
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Design Consideration](#design-consideration)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+  - [NodeResourcesFitPlus](#noderesourcesfitplus)
+  - [ScarceResourceAvoidance](#scarceresourceavoidance)
+- [Example](#example)
+  - [Scheduler Configuration](#scheduler-configuration)
+  - [GPU](#gpu)
+  - [CPU](#cpu)
+<!-- /toc -->
+
+
+## summary
+
+The NodeResourcesFit plug-in of native k8s can only adopt a type of strategy for all resources, such as MostRequestedPriority and LeastRequestedPriority. However, in industrial practice, this design does not apply to some scenarios. For example: In AI scenarios, businesses that apply for GPUs prefer to occupy the entire GPU machine first to prevent GPU fragmentation; businesses that apply for CPU & MEM are prioritized and dispersed to non-GPU machines to prevent excessive consumption of CPU & MEM on GPU machines, resulting in real tasks of applying for GPUs. Pending due to insufficient non-GPU resources
+. It is therefore hoped that both strategies can be extended to address this business need.
+
+## motivation
+case:
+- GPU tasks take priority over the entire GPU
+- CPU&MEM tasks are distributed to the CPU machine first
+
+## design-consideration
+
+- The solution is more versatile, not limited to AI clusters or CPU clusters, and not limited to common CPU resources or extended GPU resources.
+
+- Different resource policies can be configured for different cluster types and prioritized in the form of weights.
+
+- Easy to expand
+
+### goals
+
+- Different types of resources can be configured with different strategies to prioritize them in the form of weights
+
+- Prevent pods that have not applied for scarce resources from being scheduled to nodes with scarce resources.
+
+### non-goals
+
+- None.
+
+## proposal
+
+Extend two plug-ins to meet the above needs
+
+- NodeResourcesFitPlus
+- ScarceResourceAvoidance
+
+## design-details
+
+### NodeResourcesFitPlus
+config:
+```
+resources: 
+  nvidia.com/gpu:
+    type: MostAllocated
+    weight: 2
+  cpu:
+    type: LeastAllocated
+    weight: 1
+  memory:
+    type: LeastAllocated
+    weight: 1
+```
+config description:
```

Comment:
recommend giving a simple example of how to be compatible with the vanilla NodeResourceFit

Reply:
What is vanilla? Can you explain it to me?

Reply:
In addition, this strategy should not be compatible with the native NodeResourceFit strategy. If you think that all resource types use the same strategy, use NodeResourceFit. If you want to treat it differently, use NodeResourceFitPlus.

Reply:
As a production-ready scheduler, it is essential for community users to dynamically switch the strategy without restarting the instance. And, it is more user-friendly to be compatible with the native NodeResourceFit.

Reply:
done